src/tests/ftest/osa/osa_online_parallel_test.py

```diff
@@ -0,0 +1,289 @@
+        target_list.append(n+1)
+        t_string = "{},{}".format(target_list[0], target_list[1])
+
+        # Exclude rank 2.
+        rank = 2
+
+        # Start the daos_racer thread
+        kwargs = {"results": self.ds_racer_queue}
+        daos_racer_thread = threading.Thread(target=self.daos_racer_thread,
+                                             kwargs=kwargs)
+        daos_racer_thread.start()
+        time.sleep(30)
+
+        for val in range(0, num_pool):
+            pool[val] = TestPool(self.context,
+                                 dmg_command=self.get_dmg_command())
+            pool[val].get_params(self)
+            # Split total SCM and NVME size for creating multiple pools.
+            pool[val].scm_size.value = int(pool[val].scm_size.value /
+                                           num_pool)
+            pool[val].nvme_size.value = int(pool[val].nvme_size.value /
+                                            num_pool)
+            pool[val].create()
+            pool_uuid.append(pool[val].uuid)
+
+        # Exclude and reintegrate the pool_uuid, rank and targets
+        for val in range(0, num_pool):
+            self.pool = pool[val]
+            self.pool.display_pool_daos_space("Pool space: Beginning")
+            pver_begin = self.get_pool_version()
+            self.log.info("Pool Version at the beginning %s", pver_begin)
+
+            for oclass, api, test, flags in product(self.ior_dfs_oclass,
+                                                    self.ior_apis,
+                                                    self.ior_test_sequence,
+                                                    self.ior_flags):
+                threads = []
+                # Action dictionary with OSA dmg command parameters
+                action_args = {
+                    "drain": {"pool": self.pool.uuid, "rank": rank,
+                              "tgt_idx": None},
+                    "exclude": {"pool": self.pool.uuid, "rank": (rank + 1),
+                                "tgt_idx": t_string},
+                    "reintegrate": {"pool": self.pool.uuid, "rank": (rank + 1),
+                                    "tgt_idx": t_string}
+                }
+                for _ in range(0, num_jobs):
+                    # Add a thread for these IOR arguments
+                    threads.append(threading.Thread(target=self.ior_thread,
+                                                    kwargs={"pool": pool[val],
+                                                            "oclass": oclass,
+                                                            "api": api,
+                                                            "test": test,
+                                                            "flags": flags,
+                                                            "results":
+                                                            self.out_queue}))
+                for action in sorted(action_args.keys()):
```

Comment:
(pylint-consider-iterating-dictionary) Consider iterating the dictionary directly instead of calling .keys()

Reply:
Done.